File: solutions/1781. Sum of Beauty of All Substrings/1781.cpp
// Time complexity: O(n^2)
// Space complexity: O(1)
class Solution {
 public:
  int beautySum(string_view s) {
    int ans = 0;

    for (int i = 0; i < s.length(); ++i) {
      vector<int> count(26);
      for (int j = i; j < s.length(); ++j) {
        ++count[s[j] - 'a'];
        ans += ranges::max(count) - getMinFreq(count);
      }
    }

    return ans;
  }
 private:
  int getMinFreq(const vector<int>& count) {
    int minFreq = INT_MAX;
    for (const int freq : count)
      if (freq > 0)
        minFreq = min(minFreq, freq);

    return minFreq;
  }
};

File: solutions/1781. Sum of Beauty of All Substrings/1781.rs
// Time complexity: O(n^2)
// Space complexity: O(1)
impl Solution {
    pub fn beauty_sum(s: String) -> i32 {
        let mut ans = 0;

        for i in 0..s.len() {
            let mut count = vec![0; 26];
            for j in i..s.len() {
                count[(s.as_bytes()[j] - b'a') as usize] += 1;
                ans +=
                    count.iter().max().unwrap() - count.iter().filter(|&&x| x > 0).min().unwrap();
            }
        }

        ans
    }
}
